packages/app/src/cli/models/app/app.ts

```diff
@@ -7,6 +7,7 @@ import {getDependencies, PackageManager, readAndParsePackageJson} from '@shopify
 
 export const AppConfigurationSchema = schema.define.object({
   scopes: schema.define.string().default(''),
+  additionalExtensionDirectories: schema.define.array(schema.define.string()).default([]),
```

Comment:
Purely additive configuration is a very unusual concept, and IMO misses the point for developers who want control over their projects. Having configuration that is only partially overridable by developers is bound to get in the way. It also forces us to choose complex and ugly configuration options; `extensions = ["extensions/*"]` is so much nicer, and mirrors what they would see in defining Yarn/NPM/PNPM workspace configurations.

Reply:
I am not sure the link you posted is really relevant — the conventions for configuration as code vs static configuration are naturally going to be a little different. That said, even in the example you posted, presumably setting an array does not merge your options in with Rails' defaults.

I think we need to do a lot less assuming what developers *should* do. We have almost no apps that use extensions in production at Shopify; developers know **much** better than us what they need to be doing to make use of the tools we give them. I imagine you have seen a lot of the same feedback I have in checkout, but from my perspective it has been very clear: developers need flexibility. They were stuck because of us sticking to `web/*` for too long, and many of them are stuck because they have a good reason for separating extensions into multiple places/ repos.

Reply:
I am honestly struggling to see how we could prefer a developer write this configuration:

```toml
# Merges with ["extensions/*"], which you can't override
additional_extension_directories = ["my-extensions/*"]
```

over this one:

```toml
# Only the extensions listed will count
extensions = ["my-extensions/*"]
```

They both allow you to omit the field entirely to get the default `extensions = ["extensions/*"]` option; the second one is significantly easier to read, understand, and document, and leaves the developer in full control when they are taking the wheel.